include/revng/PipelineC/PipelineC.h

```diff
@@ -0,0 +1,378 @@
+
+/**
+ * serialize the provided container at the provided step in the provided path
+ *
+ * no argument can be null
+ *
+ * returns 0 if an error was encountered 1 otherwise
+ */
+bool rp_container_checkpoint_store(rp_container_checkpoint *container,
+                                   const char *path);
+
+/**
+ * load the provided container at the provided step from the provided path
+ *
+ * no argument can be null
+ *
+ * return 0 if a error was encountered 1 otherwise
+ *
+ */
+bool rp_container_checkpoint_load(rp_container_checkpoint *container,
+                                  const char *path);
+
+/**
+ * returns a table containing all the scalar available target that can
+ * be requested to the runner
+ *
+ * runner cannot be null
+ */
+void rp_runner_recompute_all_available_targets(rp_runner *runner);
+
+/**
+ * returns the container status associated to the provided checkpoint
+ * or null if no status is associated to the provided checkpoint
+ *
+ * no argument can be null
+ */
+rp_container_status *
+rp_get_container_status(rp_runner *runner, rp_container_checkpoint *container);
+
+/**
+ * returns the amount of target inside the provided container statuses
+ */
+uint64_t rp_container_status_count(rp_container_status *status);
+
+/**
+ *  returns the nth target inside the provided status or null if it's out of
+ * bounds
+ *
+ *  status cannot be null
+ */
+rp_target *
+rp_container_status_get_target(rp_container_status *status, uint64_t index);
+
+/**
+ * returns the kind of the provided target
+ *
+ * target cannot be null
+ */
+rp_kind *rp_target_get_kind(rp_target *target);
+
+/**
+ *
+ * Serializes target into a string, target cannot be null.
+ *
+ */
+char *rp_create_target_serialization(rp_target *target);
+
+/**
+ *
+ * Deserialize a target from a string, arguments cannot be null.
+ *
+ * Returns null if the input string was malformed.
+ *
+ */
+rp_target *
+rp_create_target_from_serialization(rp_runner *runner, const char *serialized);
+
+/**
+ * returns 1 if target is requiring exactly a particular kind
+ * returns 0 if target can accept derived from targets
+ *
+ * target cannot be null
+ */
+bool rp_target_is_exact(rp_target *target);
+
+/**
+ * return the number of path component inside the target
+ *
+ * target cannot be null
+ */
+uint64_t rp_target_path_components_count(rp_target *target);
+
+/**
+ * returns the name of the nth path_component, or "*" if it represents all
+ * possible targets. returns null if index is out of bound
+ *
+ * target cannot be null
+ */
+const char *rp_target_get_path_component(rp_target *target, uint64_t index);
+
+/**
+ * returns the path of the provided container at the provided step
+ */
+char *rp_create_container_path(rp_runner *runner,
+                               const char *step_name,
+                               const char *container_name);
+
+/**
+ * frees the string returned by rp_create_container_path
+ */
+void destroy_container_path(char *container_path);
```

Comment:
```suggestion
void rp_destroy_container_path(char *container_path);
```

Reply:
because this will free the container path returned by rp_create_container_path. All functions named rp_create_x return stuff that must be freed by the user.